internal/manifests/manifests.go

```diff
@@ -0,0 +1,219 @@
+	if apierr != nil {
+		return common.GenerateErrorResponder(apierr)
+	}
+
+	if params.Folder == nil {
+		defaultFolder := models.CreateManifestParamsFolderManifests
+		params.Folder = &defaultFolder
+	}
+	fileName := filepath.Join(*params.Folder, params.FileName)
+	objectName := GetManifestObjectName(*cluster.ID, fileName)
+	exists, err := m.objectHandler.DoesObjectExist(ctx, objectName)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to delete cluster manifest %s from s3", objectName)
+		return common.GenerateErrorResponder(err)
+	}
+
+	if !exists {
+		log.Infof("Cluster manifest %s doesn't exists in s3 for cluster %s", fileName, cluster.ID)
+		return operations.NewDeleteClusterManifestOK()
+	}
+
+	if err := m.objectHandler.DeleteObject(ctx, objectName); err != nil {
+		return common.GenerateErrorResponder(fmt.Errorf("failed to delete %s from s3", objectName))
+	}
+
+	log.Infof("Done deleting cluster manifest %s for cluster %s", fileName, cluster.ID)
+	return operations.NewDeleteClusterManifestOK()
+}
+
+func (m *Manifests) DownloadClusterManifest(ctx context.Context, params operations.DownloadClusterManifestParams) middleware.Responder {
+	log := logutil.FromContext(ctx, m.log)
+	if params.Folder == nil {
+		defaultFolder := models.CreateManifestParamsFolderManifests
+		params.Folder = &defaultFolder
+	}
+	fileName := filepath.Join(*params.Folder, params.FileName)
+	log.Infof("Downloading manifest %s from cluster %s", fileName, params.ClusterID)
+
+	cluster, apierr := GetCluster(ctx, m.log, m.db, params.ClusterID.String())
+	if apierr != nil {
+		return apierr
+	}
+
+	objectName := GetManifestObjectName(*cluster.ID, fileName)
+	exists, err := m.objectHandler.DoesObjectExist(ctx, objectName)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to download cluster manifest")
+		return common.GenerateErrorResponder(err)
+	}
+
+	if !exists {
+		msg := fmt.Sprintf("Cluster manifest %s doesn't exist in cluster %s", fileName, cluster.ID)
+		log.Warn(msg)
+		return common.GenerateErrorResponderWithDefault(errors.New(msg), http.StatusNotFound)
+	}
+
+	respBody, contentLength, err := m.objectHandler.Download(ctx, objectName)
+	if err != nil {
+		log.WithError(err).Errorf("failed to download file %s from cluster: %s", params.FileName, params.ClusterID.String())
+		return common.GenerateErrorResponder(err)
+	}
+
+	return filemiddleware.NewResponder(operations.NewDownloadClusterManifestOK().WithPayload(respBody), params.FileName, contentLength)
+}
+
+// GetManifestObjectName returns the manifest object name as stored in S3
+func GetManifestObjectName(clusterID strfmt.UUID, fileName string) string {
+	return filepath.Join(string(clusterID), manifestFolder, fileName)
+}
+
+// GetCluster returns the cluster entity fetched from the DB or an error if failed to
+func GetCluster(ctx context.Context, logger logrus.FieldLogger, db *gorm.DB, clusterID string) (*common.Cluster, *common.ApiErrorResponse) {
```

Comment:
I recommend to put this function in common. can resolve a lot of duplication in the code.